plugins/VLQAna.cc

```diff
@@ -289,63 +295,102 @@ bool VLQAna::filter(edm::Event& evt, const edm::EventSetup& iSetup) {
   std::unique_ptr<vlq::Jet> theTop ;
   std::unique_ptr<vlq::Jet> theAntiHiggs ; 
   std::unique_ptr<vlq::Jet> theAntiTop ;
-  if (nTop > 0) {
-
-    theTop = std::unique_ptr<vlq::Jet>(new vlq::Jet(goodTopTaggedJets.at(0))) ; 
 
-    if (nHiggs > 0) {
-      for (vlq::Jet& hjet : goodHTaggedJets) {
-        double dphi = abs((theTop->getP4()).DeltaPhi(hjet.getP4())) ;  
-        if (dphi > 2.0) {
-          theHiggs = std::unique_ptr<vlq::Jet>(new vlq::Jet(hjet)) ; 
-          isRegionD = true ; 
-          break ; 
+  if (nTop > 0) {
```

Comment:
This manipulation for the ABCD region is best done in the anaTp.py
In that case, you just store the collections for top, anti-top, Higgs, anti-Higgs, Z, anti-Z jets in the tree.